include/time.h

```diff
@@ -65,6 +65,27 @@ struct tm {
     int     tm_isdst;
 };
 
+#if defined(__APPLE2__)
+struct prodos_datetime {
+    struct {
+        unsigned day  :5;
+        unsigned mon  :4;
+        unsigned year :7;
+    }                 date;
+    struct {
+        unsigned char min;
+        unsigned char hour;
+    }                 time;
+};
+
+int __fastcall__ prodos_datetime_to_posix_tm(struct tm *tm, struct prodos_datetime *prodos_dt);
+/* Converts a ProDOS date/time structure to a struct tm */
+
+time_t __fastcall__ prodos_datetime_to_posix_time_t(void *prodos_date_ptr);
+/* Converts a ProDOS date/time structure to a time_t UNIX timestamp */
+
+#endif
```

Comment:
I wanted to originally, but then we'd have to include <apple2.h> in stat.h, dirent.h etc, and <time.h> in apple2.h, that seemed more gross in the end?

Reply:
1. I don't think so. Headers including other headers seems more normal to me than putting "all into one" header :smirk: I propose to `#include <target.h>` where necessary.
2. https://github.com/cc65/cc65/blob/master/include/apple2.h#L153-L167 was **NO** struct type declaration. It was a variable declaration. You can't just remove it.

Reply:
1. Addressed :)
2. 3.: Indeed! I read that too fast.

Reply:
About the `extern struct datetime _datetime`: I have put it back. It references __datetime from open.s and could indeed be used by cc65 users; however, it's not used internally and not mentioned in the docs. 
The fact I've put it back using the struct datetime definition instead of the "one shot" definition means its fields changed, from `_datetime.createdate/_datetime.createtime` to `_datetime.date/_datetime.time`.
Is this acceptable?

Reply:
Thanks! That's exactly how I envisioned it. I believe this incompatible change is to be accepted. I don't see it being used a lot because of the missing documentation. Maybe you would want to add some documentation in another change(?).